### PR TITLE
GOB: Improve GobConsole

### PR DESCRIPTION
**Pull request by BJNFNE**

### CODE REVIEW

File: engines/gob/console.cpp [hunk 1]
```diff
@@ -56,6 +58,28 @@ void GobConsole::unregisterCheater() {
...
+	debugPrintf(" debugflag_enable - Enables a debug flag\n");
+	debugPrintf(" debugflag_disable - Disables a debug flag\n");
+	debugPrintf("\n");
+	debugPrintf("Commands\n");
+	debugPrintf("--------\n");
+	debugPrintf(" continue - returns back to the game\n");
+	debugPrintf(" easteregg - shows you some funny stuff\n");
+	debugPrintf(" listArchives - shows which Archives are currently being used\n");
+	debugPrintf(" cheat - enables Cheats for Geisha\n");
+	debugPrintf(" varSize - shows you the size of the space variable in bytes\n");
```
Comment (bluegr) [inline engines/gob/console.cpp:74]:
```suggestion
	debugPrintf(" varSize - shows you the size of a variable in bytes\n");
```

File: engines/gob/console.cpp [hunk 2]
```diff
@@ -56,6 +58,28 @@ void GobConsole::unregisterCheater() {
...
+	debugPrintf("Debug flags\n");
+	debugPrintf("-----------\n");
+	debugPrintf(" debugflag_list - Lists the available debug flags and their status\n");
+	debugPrintf(" debugflag_enable - Enables a debug flag\n");
+	debugPrintf(" debugflag_disable - Disables a debug flag\n");
+	debugPrintf("\n");
+	debugPrintf("Commands\n");
+	debugPrintf("--------\n");
+	debugPrintf(" continue - returns back to the game\n");
+	debugPrintf(" easteregg - shows you some funny stuff\n");
```
Comment (bluegr) [inline engines/gob/console.cpp:71]:
```suggestion
	debugPrintf(" easteregg - shows easter eggs\n");
```

File: engines/gob/console.cpp [hunk 3]
```diff
@@ -192,4 +216,17 @@ bool GobConsole::cmd_listArchives(int argc, const char **argv) {
 	return true;
 }
 
+bool GobConsole::cmd_EasterEgg(int argc, const char **argv) {
+	debugPrintf("Adibou 2\n");
+	debugPrintf("-----------\n");
+	debugPrintf("When creating a character with a name from the Coktel team (e.g. 'Cathy'), the birth date is automatically filled with the real world value.\n");
```
Comment (bluegr) [inline engines/gob/console.cpp:222]:
```suggestion
	debugPrintf("When creating a character with a name from the Coktel team (e.g. 'Cathy'), the birth date is automatically filled with their actual date of birth.\n");
```

File: engines/gob/console.cpp [hunk 4]
```diff
@@ -192,4 +216,17 @@ bool GobConsole::cmd_listArchives(int argc, const char **argv) {
...
+	debugPrintf("-----------\n");
+	debugPrintf("Adibou 1\n");
+	debugPrintf("Did you know when you press on the Sun make it dark in the German version of Adibou 1 that then the Cat song plays?\n");
```
Comment (bluegr) [inline engines/gob/console.cpp:225]:
```suggestion
	debugPrintf("When the cat song plays in the German version, you can press on the Sun and change the scene to night mode\n");
```

File: engines/gob/console.cpp (hunk 3 above)
Comment (bluegr) [inline engines/gob/console.cpp:222]: It would be good to sort these alphabetically (i.e. move the Adibou 1 entries before the Adibou 2 ones)

**Comment by BJNFNE** [comment]
@bluegr i applied the changes

**Comment by sev-** [comment]
Shouldn't easter eggs better go to our Wiki?

**Comment by BJNFNE** [comment]
> Shouldn't easter eggs better go to our Wiki?

Sure, i can do that. I would add the Easter eggs to the Game related wiki entries.

**Comment by BJNFNE** [comment]
@sev- Changes were applied, i have moved the Easter eggs to our Wiki to the Game related Wiki entries, this is now ready to be merged.

File: engines/gob/console.cpp [hunk 5]
```diff
@@ -75,6 +97,9 @@ bool GobConsole::cmd_dumpVars(int argc, const char **argv) {
 	file.flush();
 	file.close();
 
+	const char *variablesdmp = "variables.dmp";
```
Comment (bluegr) [inline engines/gob/console.cpp:100]: This should be moved above, and used in `file.open()`

**Comment by sev-** [comment]
> @sev- Changes were applied, i have moved the Easter eggs to our Wiki to the Game related Wiki entries, this is now ready to be merged.

You mean "ready to be reviewed" perhaps 😄

File: engines/gob/console.cpp [hunk 6]
```diff
@@ -67,8 +89,11 @@ bool GobConsole::cmd_dumpVars(int argc, const char **argv) {
 
 	Common::DumpFile file;
 
-	if (!file.open("variables.dmp"))
-		return true;
+    if (!file.open("variables.dmp")) {
+        const char *variablesdmp = "variables.dmp";
+        debugPrintf("Dumped %s successfully to ScummVM directory\n", variablesdmp);
+        return true;
+    }
```
Comment (bluegr) [inline engines/gob/console.cpp:96]: This has been edited incorrectly. What *should* be done is something like this:

```suggestion
    const char *variablesdmp = "variables.dmp";
    if (!file.open(variablesdmp ))
        return true;
    
    // ... dumping code remains the same ...
   
    debugPrintf("Dumped %s successfully to ScummVM directory\n", variablesdmp);
```

Comment (BJNFNE) [reply engines/gob/console.cpp]: fixed and commited

**Comment by BJNFNE** [comment]
I am done on this PR, this can be reviewed, i should have addressed everything

**Comment by bluegr** [comment]
The individual changes are OK now, however, it would be best to perform the following:
- Split the detection related commits into a separate branch / pull request
- Squash all the console related commits in one

If you have trouble splitting the commits, I can merge them as described above during the weekend

**Comment by BJNFNE** [comment]
> The individual changes are OK now, however, it would be best to perform the following:
> 
> - Split the detection related commits into a separate branch / pull request
> 
> - Squash all the console related commits in one
> 
> 
> 
> If you have trouble splitting the commits, I can merge them as described above during the weekend

Hi, could you please then help me splitting/squashing them to be able to merge, i dont think i would have time to do that in the week, so it would be great if you can help me out in the weekend with that. Thank you

**Comment by BJNFNE** [comment]
i now squashed all GobConsole related commits into one.
Regarding howto move the GOB2 and Adi2 entries i would need help to how to move them over to secound PR/branch

**Comment by BJNFNE** [comment]
Done, i cherry-picked the GOB Detection commits to the new PR #5839

**Comment by bluegr** [comment]
Thanks

Pushed the relevant commit manually: 230e89243dc44cfca5d133f4a4aa5cd6340a7519

Closing

**Pull request closed without merging**